src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -451,11 +467,38 @@ public static TargetsToPropagate FromProjectAndEntryTargets(ProjectInstance proj
 
             public ImmutableList<string> GetApplicableTargetsForReference(ProjectInstance reference)
             {
-                return (GetProjectType(reference)) switch
+                ImmutableList<string> RemoveNonexistentTargetsIfSkippable(ImmutableList<TargetSpecification> targets)
+                {
+                    var targetsToKeep = new List<string>();
+                    bool getTargetFrameworksRemoved = false;
+                    foreach (TargetSpecification target in targets)
+                    {
+                        // Keep targets that are non-skippable or that exist but are skippable.
+                        if (!target.SkipIfNonexistent || reference.Targets.ContainsKey(target.Target))
+                        {
+                            targetsToKeep.Add(target.Target);
+                        }
+                        else if (target.Target.Equals("GetTargetFrameworks"))
```

Comment:
This shouldn't be hard-coded

Reply:
`GetTargetFrameworksWithPlatformForSingleTargetFramework` below also seems pretty dubious.

Reply:
Probably the p2p protocol isn't expressed correctly.

Reply:
OK, looking at https://github.com/dotnet/msbuild/pull/5657, I think the best solution here is to actually add `SkipNonexistentTargets='true'` to `GetTargetFrameworksWithPlatformForSingleTargetFramework` in `Microsoft.Managed.After.targets`. The only issue with this is that if an inner build doesn't have `GetTargetFrameworksWithPlatformForSingleTargetFramework` defined, the outer build's `GetTargetFrameworks` target will fail since it depends on it. For non-multitargeting nodes, explicitly specifying `GetTargetFrameworksWithPlatformForSingleTargetFramework` is unnecessary since `GetTargetFrameworks` calls it, so there's no harm in marking it as `SkipNonexistentTargets`.

Reply:
Make sure to add a comment in `Microsoft.Managed.After.targets` explaining that since it doesn't strictly follow the actual behavior. Mention that the p2p protocol isn't expressive enough to specify specific targets executing from the outer to the inner build or whatever, as I believe that's the main problem here?